### PR TITLE
add CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by msehnout**
The file contains recommendations for external contributors as well as
definition of how the code changes should look like for all of us.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+$ git clone https://github.com/osbuild/osbuild-composer.git
+$ cd osbuild-composer
+$ dnf group install 'RPM Development Tools' # Install rpmbuild
+$ dnf builddep golang-github-osbuild-composer.spec # Install build-time dependencies
+$ make rpm
+$ sudo dnf install rpmbuild/RPMS/x86_64/* -y
+$ sudo dnf install cockpit-composer -y
+$ sudo systemctl enable --now osbuild-composer.socket
+$ sudo systemctl enable --now cockpit.socket
+```
```
Comment (dvdhrm) [inline CONTRIBUTING.md]: Can we split this into system-setup, repo-setup, and then the recurring build-run-cycle? Something like this:



Clone the repository and create a local checkout:

```
$ git clone https://github.com/osbuild/osbuild-composer.git
$ cd osbuild-composer
```

To install the build-requirements for Fedora and friends, use:

```
$ dnf group install 'RPM Development Tools' # Install rpmbuild
$ dnf builddep golang-github-osbuild-composer.spec # Install build-time dependencies
$ sudo dnf -y install cockpit-composer # Optional: Install cockpit integration
$ sudo systemctl start cockpit.socket # Optional: Start cockpit
```

Finally, use the following to compile the project from the working directory, install it, and then run it:

```
$ make rpm
$ sudo dnf -y install rpmbuild/RPMS/x86_64/*
$ sudo systemctl start osbuild-composer.socket
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+$ git clone https://github.com/osbuild/osbuild-composer.git
+$ cd osbuild-composer
+$ dnf group install 'RPM Development Tools' # Install rpmbuild
+$ dnf builddep golang-github-osbuild-composer.spec # Install build-time dependencies
+$ make rpm
+$ sudo dnf install rpmbuild/RPMS/x86_64/* -y
+$ sudo dnf install cockpit-composer -y
+$ sudo systemctl enable --now osbuild-composer.socket
+$ sudo systemctl enable --now cockpit.socket
```
Comment (dvdhrm) [inline CONTRIBUTING.md]: Why `enable`? Shouldn't this be `start`?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+You can now open https://localhost:9090 in your browser and check the "Image Builder" section.
```
Comment (dvdhrm) [inline CONTRIBUTING.md]: I think we should mention here that this is cockpit running.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+```
+$ go test ./...
+$ go build ./...
+$ go run ./cmd/osbuild-pipeline/
+```
+
+### Testing
+
+See [this document](test/README.md) for more information about testing.
```
Comment (dvdhrm) [inline CONTRIBUTING.md]: -`this document` +`README.md`?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+### Creating a PR
+
+The commits in the PR should have these properties:
+
+* Preferably minimal and well documented
+  * Where minimal means: don't do unrelated changes even if the code is obviously wrong
+  * Well documented: both code and commit message
+  * The commit message should start with the module you work on, like: `weldr: `, or `rcm:`
+* The code should compile and the composer should start, so that we can run `git bisect` on it
+* All code should be covered with unit-tests
```
Comment (dvdhrm) [inline CONTRIBUTING.md]: -`with` +`by`, otherwise you ask for unit-tests all over the code ;)

File: CONTRIBUTING.md (hunk 1 above)
Comment (msehnout) [reply CONTRIBUTING.md]: sounds good, thanks!

File: CONTRIBUTING.md (hunk 2 above)
Comment (msehnout) [reply CONTRIBUTING.md]: `--now` makes it start right away, but it is mostly a personal preference, I want it to start immediatelly after rebooting.

File: CONTRIBUTING.md (hunk 5 above)
Comment (msehnout) [reply CONTRIBUTING.md]: hehe, what a lovely difference :)

**Comment by msehnout** [comment]
> I like it! Let's get this in. I have a few suggestions, but I can also edit it myself once merged. Up to you!

Thanks @dvdhrm , I've included all the changes you suggested.

File: CONTRIBUTING.md (hunk 4 above)
Comment (msehnout) [reply CONTRIBUTING.md]: I think +`test/README.md` makes more sense here, beacuse we also have `README.md` in the root directory and that is not the one I refer to.

**Pull request merged**